examples/make_event_id.py

```diff
@@ -0,0 +1,39 @@
+"""
+===========================
+Make an event id for Epochs
+===========================
+
+Combine triggers into a single event id while maintaining their value.
+"""
+# Author: Teon Brooks <teon.brooks@gmail.com>
+#
+# License: BSD (3-clause)
+
+import mne
+from mne.datasets import sample
+
+print(__doc__)
+
+data_path = sample.data_path()
+fname = data_path + '/MEG/sample/sample_audvis_raw-eve.fif'
+raw_fname = data_path + '/MEG/sample/sample_audvis_raw.fif'
+tmin, tmax = -1, .2
+
+# Reading events
+events = mne.read_events(fname)
+
+# Combine triggers across
+event_id = {'aud-l': 1,
+            'aud-r': 2,
+            'vis-l': 3,
+            'vis-r': 4,
+            'aud': [1, 2],
+            'vis': [3, 4]}
```

Comment:
Seems this is more flexible, it would allow you generate arbitarary and
non-hrierarchical combinations, e.g., interactions as subset. More
complementary as I see it.

2015-04-20 12:14 GMT+02:00 Alexandre Gramfort <notifications@github.com>:

> In examples/make_event_id.py
> <https://github.com/mne-tools/mne-python/pull/1976#discussion_r28677293>:
>
> > +
> > +data_path = sample.data_path()
> > +fname = data_path + '/MEG/sample/sample_audvis_raw-eve.fif'
> > +raw_fname = data_path + '/MEG/sample/sample_audvis_raw.fif'
> > +tmin, tmax = -1, .2
> > +
> > +# Reading events
> > +events = mne.read_events(fname)
> > +
> > +# Combine triggers across
> > +event_id = {'aud-l': 1,
> > +            'aud-r': 2,
> > +            'vis-l': 3,
> > +            'vis-r': 4,
> > +            'aud': [1, 2],
> > +            'vis': [3, 4]}
>
> to me this is redundant. The option aud/l, aud/r that Jona implements
> allows you to do the same. Or am I missing something?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/mne-tools/mne-python/pull/1976/files#r28677293>.
>